test-utils/state-viewer/src/epoch_info.rs

```diff
@@ -0,0 +1,213 @@
+    epoch_id: Option<EpochId>,
+    epoch_height: Option<EpochHeight>,
+    block_hash: Option<CryptoHash>,
+    block_height: Option<BlockHeight>,
+    protocol_version: Option<ProtocolVersion>,
+    store: Arc<Store>,
+    chain_store: &mut ChainStore,
+    epoch_manager: &mut EpochManager,
+) -> Vec<EpochId> {
+    if let Some(epoch_id) = epoch_id {
+        // Fetch the specified epoch.
+        vec![epoch_id]
+    } else if let Some(epoch_height) = epoch_height {
+        // Fetch epochs at the given height.
+        // There should only be one epoch at a given height. But this is a debug tool, let's check
+        // if there are multiple epochs at a given height.
+        iterate_and_filter(store, |epoch_info| epoch_info.epoch_height() == epoch_height)
+    } else if let Some(block_hash) = block_hash {
+        // Fetch an epoch containing the given block.
+        vec![epoch_manager.get_block_info(&block_hash).unwrap().epoch_id().clone()]
+    } else if let Some(block_height) = block_height {
+        // Fetch an epoch containing the given block height.
+        let block_hash = chain_store.get_block_hash_by_height(block_height).unwrap();
+        vec![epoch_manager.get_block_info(&block_hash).unwrap().epoch_id().clone()]
+    } else if let Some(protocol_version) = protocol_version {
+        // Fetch the first epoch of the given protocol version.
+        iterate_and_filter(store, |epoch_info| epoch_info.protocol_version() == protocol_version)
+    } else {
+        // Fetch all epochs.
+        iterate_and_filter(store, |_| true)
+    }
+}
+
+// Iterates over the ColEpochInfo column, ignores AGGREGATOR_KEY and returns deserialized EpochId
+// for EpochInfos that satisfy the given predicate.
+fn iterate_and_filter(store: Arc<Store>, predicate: impl Fn(EpochInfo) -> bool) -> Vec<EpochId> {
+    store
+        .iter(DBCol::ColEpochInfo)
+        .filter_map(|(key, value)| {
+            if key.as_ref() == AGGREGATOR_KEY {
+                None
+            } else {
+                let epoch_info = EpochInfo::try_from_slice(value.as_ref()).unwrap();
+                if predicate(epoch_info) {
+                    Some(EpochId::try_from_slice(key.as_ref()).unwrap())
+                } else {
+                    None
+                }
+            }
+        })
+        .collect()
+}
+
+fn display_epoch_info(
+    epoch_id: &EpochId,
+    epoch_info: &EpochInfo,
+    validator_account_id: &Option<AccountId>,
+    head_epoch_height: &EpochHeight,
+    chain_store: &mut ChainStore,
+    epoch_manager: &mut EpochManager,
+    runtime_adapter: Arc<dyn RuntimeAdapter>,
+) {
+    println!("{:?}: {:#?}", epoch_id, epoch_info);
+    if epoch_info.epoch_height() >= *head_epoch_height {
+        println!("Epoch information for this epoch is not yet available, skipping.");
+        return;
+    }
+    if let Some(account_id) = validator_account_id.clone() {
+        if let Some(kickout) = epoch_info.validator_kickout().get(&account_id) {
```

Comment:
optional: I'd refactor 167- 212 into a separate method (that displays info from specific validator point of view)